app/controllers/partners/referrals_controller.rb

```diff
@@ -0,0 +1,61 @@
+module Partners
+    class ReferralsController < ApplicationController
+    include PromosHelper
+
+    def index
+        render "referrals"
+    end
+
+    def data
```

Comment:
Instead of this being a separate route that doesn't really conform to anything let's move it to the index. You can respond to different formats like so

```ruby
respond_to do |format|
  format.html
  format.json  do 
    render json: method_producing_data
  end 
end
```